Added APIv3 document thingo

ocket8888 opened the pull request:
Adds a LaTeX (XeTeX with minted) document describing a plan for Traffic Ops API v3.0 in terms of guidelines and principles, primary objects, and endpoint specifications.

ocket8888 committed:
Added APIv3 document thingo

rob05c commented on blueprints/APIv3.tex:
```diff
@@ -0,0 +1,507 @@
+change only part of a stored object. When an object is modified, the response body MUST contain a representation of the object after modification,
+and that representation SHOULD fully describe the modified object, even the parts that were not modified.\\
+Handlers that implement PATCH in the Traffic Ops API MUST use conditional requests to ensure that race conditions are not a problem, e.g. using
+ETag and If-Match, or If-Not-Modified-Since.\\
```
Should be `If-Unmodiifed-Since`. See https://tools.ietf.org/html/rfc7232#section-3.4
Also, consider adding `If-None-Match`. It's less commonly known, but much better than IMS. IMO it'd be good to list as a hard requirement as well.

ocket8888 replied:
I put the hard requirements in a different section, those were just examples. Although, now that it's a hard requirement, maybe I don't need this restriction here?

ocket8888 committed:
Added proper type/representation of empty collection properties

ocket8888 committed:
Added note about request path suffixes

ocket8888 committed:
Added definition of tenantable objects

ocket8888 committed:
Added build artifacts to gitignore

ocket8888 committed:
Added admonition about no duplicate endpoint functionality

Also relaxed restriction on path parameters. I realize those aren't totally compatible
but a discussion about that is upcoming.
